lib/Parse/ParsePattern.cpp

```diff
@@ -366,7 +366,13 @@ Parser::parseParameterClause(SourceLoc &leftParenLoc,
       } else {
         // Otherwise, we're not sure what is going on, but this doesn't smell
         // like a parameter.
-        diagnose(Tok, diag::expected_parameter_name);
+        if (Tok.isBinaryOperator() && Tok.getText() == "==") {
+          diagnose(Tok,
+                   diag::expected_assignment_instead_of_comparison_operator)
+              .fixItReplace(Tok.getLoc(), "=");
```

Comment:
Is it possible for the compiler to check this earlier and recover by parsing the expression normally?

Reply:
The check could indeed be taken before l.272 for instance, but just to be sure I understand: in this case "recover" means that the token for '==' should be consumed so that the code can continue parsing the default value and so on?

Reply:
Yeah, I was thinking you'd essentially have the compiler just pretend that the user wrote `=`.

Reply:
Got it. Will try this.

Reply:
I think this check and recovery should be performed *later*, like l.385.

Reply:
but somehow I need to avoid the parsing logic to get into any of this [branches](https://github.com/apple/swift/blob/04ca9752cf4b5e2700b6663d0ddd381ea0fed3ab/lib/Parse/ParsePattern.cpp#L273-L376), so a check would be needed before.

Reply:
It should work as expected if you consume `==` as if it's `=` and parse the default value after `==`.

Reply:
pushed https://github.com/apple/swift/pull/26139/commits/d82dd84592c05655faa3e1e8fde3d6b038a59225 so check is now performed later as @rintaro suggested and compiler is recovering :).